src/maputil.rs

```diff
@@ -11,31 +12,35 @@ pub enum Tile<'a> {
     SnakeBody { coordinate: (i32,i32), snake: &'a SnakeInfo }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Copy, Debug)]
 pub enum Direction {
     Down,
     Up,
     Left,
     Right
 }
 
-pub fn direction_as_string(direction: &Direction) -> String {
-    let s = match direction {
-        &Direction::Down => "DOWN",
-        &Direction::Up => "UP",
-        &Direction::Left => "LEFT",
-        &Direction::Right => "RIGHT"
-    };
-
-    String::from(s)
+impl Serialize for Direction {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_str(match *self {
+            Direction::Down  => "DOWN",
+            Direction::Up    => "UP",
+            Direction::Left  => "LEFT",
+            Direction::Right => "RIGHT",
+        })
+    }
 }
 
-pub fn direction_as_movement_delta(direction: &Direction) -> (i32,i32) {
-    match direction {
-        &Direction::Down => (0, 1),
-        &Direction::Up => (0, -1),
-        &Direction::Left => (-1, 0),
-        &Direction::Right => (1, 0)
+impl Direction {
+    pub fn direction_as_movement_delta(&self) -> (i32,i32) {
```

Comment:
I see your point, but I think the ergonomics of `my_direction.direction_as_movement_delta()` are excellent. And I agree to 100% that `as_movement_delta()`is a better name.